extensions/s3/build.gradle

```diff
@@ -25,14 +25,14 @@ dependencies {
     Classpaths.inheritAutoService(project)
     Classpaths.inheritImmutables(project)
 
-    Classpaths.inheritJUnitPlatform(project)
+    Classpaths.inheritJUnitClassic(project, 'testImplementation')
```

Comment:
I had to change from Junit5 to Junit4 because I wanted to use the following to set up a query context for making `Table` objects in `S3ParquetTests` class
```
    @Rule
    public final EngineCleanup framework = new EngineCleanup();
```
@devinrsmith can you let me know if this is okay or there is a better way to do this?

Reply:
I'm really trying to move us forward to JUnit 5, especially with new code. If absolutely necessary, you can use JUnit 5 runner against JUnit 4 tests; I believe there may be an example or two of that we use elsewhere.

That said, I _don't_ think we should be moving engine tests into the S3 layer - I think the strategy of bringing in S3 testing to the engine (/parquet) is the better approach. I think this should be doable after #5442.